unix/vncserver/vncsession.c

```diff
@@ -511,14 +511,23 @@ main(int argc, char **argv)
 
     const char *username, *display;
 
-    if ((argc != 3) || (argv[2][0] != ':')) {
+    int forking = 1;
+    if (argc == 4) {
+        if ((getopt(argc, argv, "D") == 'D') && (argv[3][0] == ':'))
```

Comment:
Let's do a normal `getopt()` loop so that people feel familiar with the code.

Reply:
No problem.

Reply:
This last push is needed to detect some corner error cases when we should return EX_USAGE after adding the getopt() loop. Sorry about the extra noise. Now I think the PR is good.